tests/test_mpls.py

```diff
@@ -0,0 +1,1049 @@
+        dvs.servers[1].runcmd("ip route add default via 10.0.0.2")
+
+        # get neighbor and arp entry
+        dvs.servers[0].runcmd("ping -c 1 10.0.0.3")
+
+        # add route entry
+        # dvs.runcmd("ip route add 2.2.2.0/24 encap mpls 201 via inet 10.0.0.1 dev Ethernet0")
+        prefix = "2.2.2.0/24"
+        fieldValues = {"nexthop": "10.0.0.1", "ifname": "Ethernet0", "mpls_nh" : "push+201"}
+        self.create_route_entry(prefix, fieldValues)
+
+        # check application database
+        self.pdb.wait_for_entry("ROUTE_TABLE", prefix)
+
+        # check ASIC route database
+        self.check_route_entries([prefix])
+
+        # remove route entry
+        # dvs.runcmd("ip route del 2.2.2.0/24 encap mpls 201 via inet 10.0.0.1 dev Ethernet0")
+        self.remove_route_entry(prefix)
+
+        # check application database
+        self.pdb.wait_for_deleted_entry("ROUTE_TABLE", prefix)
+
+        # check ASIC route database
+        self.check_deleted_route_entries([prefix])
+
+        # remove ip address
+        self.remove_ip_address("Ethernet0", "10.0.0.0/31")
+        self.remove_ip_address("Ethernet4", "10.0.0.2/31")
+
+        # remove mpls interface
+        self.remove_mpls_intf("Ethernet0")
+        self.remove_mpls_intf("Ethernet4")
+
+        self.set_admin_status("Ethernet0", "down")
+        self.set_admin_status("Ethernet4", "down")
+
+        # remove ip address and default route
+        dvs.servers[0].runcmd("ip route del default dev eth0")
+        dvs.servers[0].runcmd("ip address del 10.0.0.1/31 dev eth0")
+
+        dvs.servers[1].runcmd("ip route del default dev eth0")
+        dvs.servers[1].runcmd("ip address del 10.0.0.3/31 dev eth0")
+
+        self.clear_mpls(dvs)
+
+    def test_RouteAddRemoveMplsSwapRoute(self, dvs, testlog):
+        self.setup_mpls(dvs)
+
+        self.setup_db(dvs)
+
+        self.clear_srv_config(dvs)
+
+        # create mpls interface
+        self.create_mpls_intf("Ethernet0")
+        self.create_mpls_intf("Ethernet4")
+
+        # set ip address
+        self.add_ip_address("Ethernet0", "10.0.0.0/31")
+        self.add_ip_address("Ethernet4", "10.0.0.2/31")
+
+        # bring up interface
+        self.set_admin_status("Ethernet0", "up")
+        self.set_admin_status("Ethernet4", "up")
+
+        # set ip address and default route
+        dvs.servers[0].runcmd("ip address add 10.0.0.1/31 dev eth0")
+        dvs.servers[0].runcmd("ip route add default via 10.0.0.0")
+
+        dvs.servers[1].runcmd("ip address add 10.0.0.3/31 dev eth0")
+        dvs.servers[1].runcmd("ip route add default via 10.0.0.2")
+
+        # get neighbor and arp entry
+        dvs.servers[0].runcmd("ping -c 1 10.0.0.3")
+
+        # add route entry
+        # dvs.runcmd("ip -f mpls route add 200 as 201 via inet 10.0.0.1 dev Ethernet0")
+        label = "200"
+        fieldValues = {"nexthop": "10.0.0.1", "ifname": "Ethernet0", "mpls_nh" : "swap+201", "mpls_pop" : "1"}
```

Comment:
can you also add a 'multi label' "swap+201/202" swap case. This is a negative test case, and should not added multi label swap as it's not supported.

Reply:
@smaheshm So orchagent should prevent multi-label NH?  Linux kernel does support multi-label NHs so the VS platform also supports this... your suggested negative testcase will not fail unless we change orchagent to do the rejection.

Reply:
yes, as per SAI API, ASICs that support SWAP can SWAP with only one label. As per MPLS RFC 'SWAP' is always with one label.